contract-message-test/tests/harness.rs

```diff
@@ -1,184 +1,659 @@
-use fuel_crypto::Hasher;
-use fuels::contract::script::Script;
-use fuels::prelude::*;
-use fuels::test_helpers::Config;
-use fuels::tx::{AssetId, Bytes32, Contract as tx_contract, Input, Output, Transaction, UtxoId};
+mod utils {
+    pub mod environment;
+    pub mod ext_fuel_core;
+    pub mod ext_sdk_provider;
+}
+use std::str::FromStr;
+
+use fuels::prelude::Contract;
+use fuels::prelude::Salt;
+use fuels::prelude::StorageConfiguration;
+use fuels::prelude::TxParameters;
+use fuels::tx::Input;
+use fuels::tx::Output;
+use fuels::tx::Transaction;
+use fuels::tx::TxPointer;
+use fuels::tx::UtxoId;
+use utils::environment as env;
+use utils::ext_fuel_core;
+use utils::ext_sdk_provider;
+
+use fuels::test_helpers::DEFAULT_COIN_AMOUNT;
+use fuels::tx::{Address, AssetId, Bytes32, ContractId, Word};
+
+pub const RANDOM_SALT: &str = "0x1a896ebd5f55c10bc830755278e6d2b9278b4177b8bca400d3e7710eee293786";
+pub const RANDOM_SALT2: &str = "0xd5f55c10bc830755278e6d2b9278b4177b8bca401a896eb0d3e7710eee293786";
+
+///////////////////
+// SUCCESS CASES //
+///////////////////
+
+#[tokio::test]
+async fn relay_message_with_predicate_and_script_constraint() {
```

Comment:
I would first separate these tests by some concept and then split that concept into a successful case and a reverting case. An example can be seen in the [apps](https://github.com/FuelLabs/sway-applications/blob/master/escrow/tests/functions/deposit.rs) where we separate by a function and then within that function we separate by success or reverts. This structures the data and allows for easier parsing of content

Reply:
There aren't any separate "functions" to split the tests up by. It's just "did the predicate validate and the script execute" or "did it not". There's just a lot of conditions in the predicate that can be tweaked, so there's a lot of possible input combinations to test. I do like the idea of splitting into two separate modules for success and panic though!